File: Exercises/101-109 - Small Problems/medium_1/word_to_digit.rb
# Write a method that takes a sentence string as input, and returns a new string that contains the original
# string with any sequence of the words 'zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight',
# 'nine' converted to a string of digits.

#Example:

# ```
# word_to_digit('Please call me at five five five one two three four. Thanks.') ==
# 'Please call me at 5 5 5 1 2 3 4. Thanks.'
# ```

DIGIT_HASH = {
  'zero' => '0', 'one' => '1', 'two' => '2', 'three' => '3', 'four' => '4',
  'five' => '5', 'six' => '6', 'seven' => '7', 'eight' => '8', 'nine' => '9'
}.freeze

def word_to_digit(words)
  DIGIT_HASH.keys.each do |word|
    words.gsub!(/\b#{word}\b/, DIGIT_HASH[word])
  end
  words
end

 puts word_to_digit('Please call me at five five five one two three four. Thanks.') ==
       'Please call me at 5 5 5 1 2 3 4. Thanks.'

File: Exercises/210 - Small Problems/medium-1/word-to-digit.js
/*
Write a function that takes a sentence string as input and returns a new string that contains the original string with any sequence of the words 'zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine' converted to a string of digits.

Example:

wordToDigit('Please call me at five five five one two three four. Thanks.');
//'Please call me at 5 5 5 1 2 3 4. Thanks.'
*/

// Create an array of strings in numerical order.
// Split the string at each space for an array of words.
// Map the words array and replace number strings with index of the array.

var DIGITS = [
  'zero', 'one', 'two', 'three', 'four', 'five',
  'six', 'seven', 'eight', 'nine', 'ten'
];

function wordToDigit(str) {
  return str.split(' ').map(function(word) {
    var punc = word.replace(/[^!?.-]+/ig, '');
    var noPunc = word.replace(/[!?.-]+$/ig, '');
    var index = DIGITS.indexOf(noPunc.toLowerCase());

    return index < 0 ? word : String(index) + punc;
  }).join(' ');
}

console.log(wordToDigit('Please call me at five five five one two three four. Thanks.'));